docs/blueprints/planned-downtime.md

```diff
@@ -0,0 +1,62 @@
+
+The main thing that needs to be added on the storagenode side is an endpoint exclusively for satellite use that allows the satellite to initiate a piece transfer request unrelated to graceful exit. It is possible that we could use the same code path that graceful exit will use (we are currently refactoring graceful exit), but this might not be the best idea. In any case, it is worth considering, especially if we want to stream many piece transfer requests back-to-back.
+
+On the satellite side, we will need to add a chore that does the following:
+1. Get a list of nodes with upcoming planned downtime within a configurable interval (TODO add list of config values to this doc)
+2. Create a metainfo loop observer which generates a list of unhealthy and/or close-to-unhealthy segments which these planned downtime nodes are also a part of (can this somehow be added to the checker?). The output of the observer should be a map of node IDs to piece IDs which need to be transferred (as well as any other important info in metainfo needed for a piece transfer). (TODO how do we limit memory usage?)
+3. Create a limiter for opening concurrent connections to nodes, and using this limiter, for each node, attempt to transfer all of the relevant pieces.
+
+An attempt to transfer a stream of pieces for a given node will look like:
+1. Open connection to stream piece transfer requests to nodes (like graceful exit, but initiated by satellite)
+2. Select target node from overlay cache. Exclude nodes with upcoming planned downtime (TODO come up with easy to say name that describes "nodes with upcoming planned downtime")
+3. Construct and sign piece transfer request with origin node, target node, piece ID, any other relevant info
+4. Send piece transfer request to node and wait for response
+5. On failure, retry (TODO config value for limiting number of retries)
+6. On success, go to next piece and repeat from step 2
+7. On list completion, close connection
+
+The chore will run on a configured interval - if there are nodes with upcoming planned downtime, it will go ahead and try to transfer pieces off of them. Otherwise, it will wait before checking again.
+
+Open questions:
+
+* How do we limit planned downtime? I see two main options (but there are probably more): 1. Have a specific number of hours per year that a node can use for planned downtime. Once they go over, they simply stop being able to schedule planned downtime. 2. No strict limit on planned downtime. Rather, financially disincentivize nodes from taking planned downtime. No disincentive for first x hours, but after that, planned downtime can reduce payout
+* What should the interface for planned downtime look like?
+* How far in advance should planned downtime be scheduled? >24 hours? >1 month? No limitation?
```

Comment:
I think the limit is necessary so that a storage node doesn't say it wants planned downtime like an hour beforehand - in that case the satellite would not have any time to transfer pieces off the node and there would be not satellite-perspective advantage to the planned downtime feature.

I do think the satellite should reserve the right to transfer pieces off a node at any time if the node has planned downtime scheduled (even if it is way in the future). That said, I think realistically, the satellite would mainly be focused on nodes that have planned downtime in the next week or so.